### DISCUSSION TITLE
Scanning standalone .efi file

### DISCUSSION THREAD

**Discussion by miakushka** [chipsec/chipsec#1300] [Q&A] [answered]
Hi everyone and thank you for doing such an amazing framework.

I am wondering if this framework can scan for vulnerabilities standalone .efi files? I have not found any mentions regarding such functionality in the documentation.

Thank you!

**Comment by frinzell** [comment, marked as answer]
Thank you!

No scanner for .efi files specifically but the [blockedlist.json](https://github.com/chipsec/chipsec/blob/master/chipsec/modules/tools/uefi/blockedlist.json) does list some GUIDs and hashes that may help.
 
There are two that may help for FW volumes:
[scan_blocked.py](https://chipsec.github.io/modules/chipsec.modules.tools.uefi.scan_blocked.html) - can scan for some known binaries within EFI FW volumes.
[scan_image.py](https://chipsec.github.io/modules/chipsec.modules.tools.uefi.scan_image.html) - can create a sort of fingerprint of a known good FW volume to compare against other volumes.

Hope this helps!

edit: clarified support

**Reply by miakushka** [reply]
Will try these ones, thank you!

**Reply by miakushka** [reply]
Checked your updated answer. Thank you anyway!
